src/main/java/core/basesyntax/strategy/StrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.operation.OperationHandler;
+import java.util.Map;
+
+public class StrategyImpl implements Strategy {
+    private Map<FruitTransaction.Operation, OperationHandler> operationOperationHandlerMap;

```

Comment:
Consider hiding the implementation details of your strategy map by using private final for the field declaration. This ensures that the map cannot be replaced after the construction of the StrategyImpl object, which aligns with the encapsulation principle.